packages/e2e-tests/specs/editor/various/change-detection.test.js

```diff
@@ -348,4 +327,23 @@ describe( 'Change detection', () => {
 		// Verify that the post is not dirty.
 		await assertIsDirty( false );
 	} );
+
+	it( 'should not prompt to confirm unsaved changes when trashing an existing post', async () => {
+		// Enter title.
+		await page.type( '.editor-post-title__input', 'Hello World' );
+
+		// Save
+		await saveDraft();
+
+		// Trash post.
+		await openDocumentSettingsSidebar();
+		await page.click( '.editor-post-trash.components-button' );
+
+		// Wait for "Saved" to confirm save complete.
+		await page.waitForSelector( '.editor-post-saved-state.is-saved' );
+
+		// Make sure redirection happens.
+		await page.waitForNavigation();
+		isCurrentURL( 'edit.php' );
```

Comment:
`isCurrentURL` returns a `boolean`, it doesn't `throw`. Should we `expect` ?

```suggestion
		expect( isCurrentURL( 'edit.php' ) ).toBe( true );
```